refiner/scripts/seed_terminology_db.py

```diff
@@ -0,0 +1,345 @@
+        """
+
+        time.sleep(self.sleep_interval)
+        response = requests.get(
+            f"{self.api_url}/{endpoint}",
+            headers=self.headers,
+            params=params or {},
+        )
+        response.raise_for_status()
+        return response.json()
+
+    def make_tes_request_from_url(self, url: str) -> dict[str, Any]:
+        """
+        Make a request to a full URL (e.g., for pagination).
+        """
+
+        time.sleep(self.sleep_interval)
+        response = requests.get(url, headers=self.headers)
+        response.raise_for_status()
+        return response.json()
+
+    def get_next_url(self, response: dict[str, Any]) -> str | None:
+        """
+        Extract the next URL for pagination.
+        """
+
+        if "link" in response:
+            for link in response["link"]:
+                if link.get("relation") == "next":
+                    return link.get("url")
+        return None
+
+    def extract_codes(
+        self, resource: dict[str, Any]
+    ) -> dict[str, list[CodeableConcept]]:
+        """
+        Extract codes from a ValueSet resource.
+
+        Args:
+            resource: FHIR ValueSet resource dictionary
+
+        Returns:
+            Dictionary mapping code types to lists of codes
+        """
+
+        codes: dict[str, list[CodeableConcept]] = {
+            "loinc": [],
+            "snomed": [],
+            "icd10": [],
+            "rxnorm": [],
+        }
+
+        compose = resource.get("compose", {})
+        for include in compose.get("include", []):
+            system_url: str = include.get("system", "")
+            concepts: list[dict[str, str]] = include.get("concept", [])
+
+            # initialize code_type to None
+            code_type: str | None = None
+
+            match system_url:
+                case system_url if "loinc.org" in system_url:
+                    code_type = "loinc"
+                case system_url if "snomed.info" in system_url:
+                    code_type = "snomed"
+                case system_url if "icd-10-cm" in system_url:
+                    code_type = "icd10"
+                case system_url if "rxnorm" in system_url.lower():
+                    code_type = "rxnorm"
+                case _:
+                    # there are many code systems we don't need that we'll simply pass on
+                    # we just want the condition groupers
+                    continue
+
+            if not concepts:
+                self.logger.error(
+                    msg=f"No concepts found for system {system_url}",
+                    extra={"code_type": code_type},
+                )
+                continue
+
+            codes[code_type].extend(
+                [
+                    CodeableConcept(
+                        code=concept["code"], display=concept.get("display", "")
+                    )
+                    for concept in concepts
+                ]
+            )
```

Comment:
How against would you be swapping this out for this?
```suggestion
for concept in concepts:
    codeable = CodeableConcept(
        code=concept["code"],
        display=concept.get("display", "")
    )
    codes[code_type].append(codeable)
```

Maybe less pythonic but it took me a minute to understand the original 😅 I think the formatting of it was also throwing me off for some reason.

Reply:
yeah, i actually like this better. do you want to try and delete your local db and run the seed script and the other check_db script to make sure the outputs line up? couldn't hurt to just do `pytest tests/` as well